internal/resolve/catalog.go

```diff
@@ -233,18 +234,50 @@ type CatalogWalkFunc func(context.Context, *catalogd.ClusterCatalog, *declcfg.De
 
 func CatalogWalker(listCatalogs func(context.Context, ...client.ListOption) ([]catalogd.ClusterCatalog, error), getPackage func(context.Context, *catalogd.ClusterCatalog, string) (*declcfg.DeclarativeConfig, error)) func(ctx context.Context, packageName string, f CatalogWalkFunc, catalogListOpts ...client.ListOption) error {
 	return func(ctx context.Context, packageName string, f CatalogWalkFunc, catalogListOpts ...client.ListOption) error {
+		l := log.FromContext(ctx)
 		catalogs, err := listCatalogs(ctx, catalogListOpts...)
 		if err != nil {
 			return fmt.Errorf("error listing catalogs: %w", err)
 		}
 
+		// Track if at least one catalog was processed (not disabled)
+		processedCatalogs := false
+		bundleFound := false
+
 		for i := range catalogs {
 			cat := &catalogs[i]
+
+			// skip catalogs with Availability set to "Disabled"
+			if cat.Spec.Availability == "" || cat.Spec.Availability == "Enabled" {
+				// Continue processing the catalog as it is enabled
+			} else if cat.Spec.Availability == "Disabled" {
+				l.Info("excluding ClusterCatalog from resolution process since it is disabled", "catalog", cat.Name)
+				continue
+			}
+
+			// Mark that we processed at least one catalog
+			processedCatalogs = true
+
 			fbc, fbcErr := getPackage(ctx, cat, packageName)
+			if fbcErr == nil && fbc != nil {
+				bundleFound = true
+			}
+
 			if walkErr := f(ctx, cat, fbc, fbcErr); walkErr != nil {
 				return walkErr
 			}
 		}
+
+		// If no catalogs were processed at all, return a 'no catalogs' error
+		if !processedCatalogs {
+			return fmt.Errorf("no enabled catalogs found for package: %s", packageName)
+		}
+
+		// Return an error if no valid bundle was found in any processed catalog
+		if !bundleFound {
+			return fmt.Errorf("no bundles found for package: %s", packageName)
+		}
```

Comment:
This code is simply the catalog walking code, and is decoupled from the concept of finding/resolving bundles. It is _used_ by the resolver.

But maybe the terminology is confusing me? We call `getPackage` which returns the entire FBC for `packageName` from the catalog, so if we keep this, it should really be called `foundPackage`.

But I'm not sure we need this variable or error at all. The walk function (`f`), receives the packageName, fbc, and fbcErr and can decide how to handle any potential result of `getPackage`.

Reply:
Yeah I agree. I'm removing the tracking variables and associated errors, the CatalogWalker function no longer makes decisions about whether any bundles were found or whether any catalogs were processed. It simply walks through the catalogs, now aware of catalog availability.